specification/data-resource-semantic-conventions.md

```diff
@@ -70,6 +70,28 @@ Labels defining a compute unit (e.g. Container, Process, Lambda Function).
 | container.image.name | Name of the image the container was built on. | `gcr.io/opentelemetry/operator` |
 | container.image.tag | Container image tag. | `0.1` |
 
+
+### Function as a Service
+
+**type:** `faas`
+
+**Description:** A serverless instance.
+
+| Label  | Description  | Example  | Required |
+|---|---|---|--|
+| faas.region| A specific geographical location where the function is executed. | `us-east-2` | Yes |
+| faas.name | The name of the function being executed. | `opentelemetry-lambda` | Yes |
+| faas.identifier | The unique name of the function being executed. <br /> For example, in AWS this field correspond to the [ARN] value, in GCP to the URI of the resource, and Azure to the [FunctionDirectory] field. | `opentelemetry-lambda` | Yes |
+| faas.version | The version of the function being executed. | `opentelemetry-latest` | No |
+| faas.provider | Provider name of the function, e.g. `aws`, `gpc`, `azure`, `serverless`, ... | `aws` | No |
+| faas.payload | The input passed to the function. | `{name: 'opentelemetry'}` | No |
+| faas.trigger | Type of the trigger that spawned the function, e.g. `http`, `S3`, `PubSub`, ... | `http` | Yes |
+| faas.instance.id | String containing the unique execution id. | `opentelemetry-0001` | Yes |
```

Comment:
Is this instance id or execution id? An instance can be reused for multiple executions. What do you do if the service provider does not make this information available to the function?

Reply:
Same comment on naming.

Reply:
This should be optional (see @dyladan 's comment).

Reply:
I didn't think about this. You right, this should be execution id.

Reply:
You still may want instance ID in the case where you save some state to a global variable which is saved between invocations. If your first invocation works, but the second one fails consistently, this might be good information to have for debugging.

Reply:
That's a good point!